src/main/webapp/app/markdown-editor/commands/bold.command.ts

```diff
@@ -15,34 +15,15 @@ export class BoldCommand extends Command {
     execute(): void {
         const selectedText = this.getSelectedText();
         let textToAdd = '';
-        let startSpace = '';
-        let endSpace = '';
-        let startIndx = 0;
-        let endIndx = 0;
-        const selectedTextLength = selectedText.length;
 
-        if (selectedText.includes('**')) {
+        if (selectedText.charAt(0) === '*' && selectedText.charAt(selectedText.length - 1) === '*') {
             textToAdd = selectedText.slice(2, -2);
             this.insertText(textToAdd);
         } else {
-            for (let i = 0; i < selectedTextLength; i++) {
-                if (selectedText.charAt(i) === ' ') {
-                    startSpace = startSpace + ' ';
-                } else {
-                    startIndx = i;
-                    break;
-                }
-            }
-
-            for (let j = selectedTextLength; j >= 0; j--) {
-                if (selectedText.charAt(j - 1) === ' ') {
-                    endSpace = endSpace + ' ';
-                } else {
-                    endIndx = j;
-                    break;
-                }
-            }
-            const newTxt = selectedText.slice(startIndx, endIndx);
+            const refinedString = this.ignoreWhiteSpace(selectedText);
```

Comment:
Except for the command characters `**`, this is has the same code as the in the `italic.command.ts` file. Maybe just move the creation of the `nexTxt` inside a new method that returns the final text to add with the command characters based on the caller.

Reply:
Thank you for your constructive feedback, I modified my code accordingly !